core/src/main/java/hudson/search/Search.java

```diff
@@ -76,7 +76,7 @@ public void doIndex(StaplerRequest req, StaplerResponse rsp) throws IOException,
                     SuggestedItem target = find(index, query, smo);
                     if(target!=null) {
                         // found
-                        rsp.sendRedirect2(a.getUrl()+target.getUrl());
+                        rsp.sendRedirect2(req.getContextPath()+target.getUrl());
```

Comment:
@jglick So I got it. However, I end up understanding that most of the tests done for `SearchTest.java` are not useful, because you are using ` Page result = j.search(projectName);` but that search method doesn't do the same that what `doIndex` does in `Search` class. So in my opinion, for example `testSearchByProjectName` is not really doing anything useful.

So main problem is that `doIndex` in `Search` class has `StaplerRequest` and `StaplerResponse` as arguments. So need to figure out how to create them for the Unit test to be able to pass them to the function. Any idea?